### ISSUE TITLE
`0f30019` breaks archive builds

### ISSUE THREAD

**Issue by 0ax1** [getsentry/sentry-cocoa#3569]
### Platform

macOS

### Environment

Develop

### Installed

Swift Package Manager

### Version

0f30019

### Did it work on previous versions?

It worked in 9e6665f, the commit before

### Steps to Reproduce

1. Create a new Xcode project.
2. Add ﻿`sentry-cocoa` as a package dependency
3. Run an archive build from the command line: `xcodebuild archive`

### Expected Result

The build succeeds.

### Actual Result

The build fails with the following error message:
```
GatherProvisioningInputs

CreateBuildDescription

Copy /tmp/abcd.dst/Applications/abcd.app/Contents/Resources/Sentry_SentryPrivate.bundle /Users/lx/Desktop/abcd/build/Release/Sentry_SentryPrivate.bundle (in target 'abcd' from project 'abcd')
    cd /Users/lx/Desktop/abcd
    builtin-copy -exclude .DS_Store -exclude CVS -exclude .svn -exclude .git -exclude .hg -resolve-src-symlinks /Users/lx/Desktop/abcd/build/Release/Sentry_SentryPrivate.bundle /tmp/abcd.dst/Applications/abcd.app/Contents/Resources
error: /Users/lx/Desktop/abcd/build/Release/Sentry_SentryPrivate.bundle: No such file or directory (in target 'abcd' from project 'abcd')
```

```
** ARCHIVE FAILED **


The following build commands failed:
        Copy /tmp/abcd.dst/Applications/abcd.app/Contents/Resources/Sentry_SentryPrivate.bundle /Users/lx/Desktop/abcd/build/Release/Sentry_SentryPrivate.bundle (in target 'abcd' from project 'abcd')
```

**Comment by philipphofmann** [comment]
@0ax1, I couldn't reproduce your issue. Can you please provide more details on your project, or upload a sample project with the issue?

**Comment by 0ax1** [comment]
@philipphofmann, thanks for looking into this! I'm on macOS 14.3 (23D56), Xcode 15.2 (15C500b).

I further attached an example project for which the archive build fails: [abcd.zip](https://github.com/getsentry/sentry-cocoa/files/14027832/abcd.zip)

**Comment by philipphofmann** [comment]
I'm sorry, but your project builds for me and the archive command doesn't fail. I'm on macOS 14.2.1 on an M2 MacBook. Do you use an Intel or Apple chip?

Please try with the latest 8.18.0 release, maybe that helps. Cleaning your build folder could also help.

**Comment by 0ax1** [comment]
@philipphofmann, thanks for trying the example project. Curious, that this succeeds on your end. I did try 8.18.0 and each other tagged version down to 8.10.0 which is the first tagged version with a successful archive build on my machine. I'm also on an M2 MacBook. I'll investigate on my end a bit and get back to you. Thanks again!

**Comment by 0ax1** [comment]
To double check: Did you run the archive command from the command line or directly from Xcode? If you ran it from Xcode directly, can you try the archive command from the CLI?

**Comment by philipphofmann** [comment]
Okay, now I can reproduce your issue. When running it from the command line with `xcodebuild archive` it doesn't work. However, I added `swift-algorithms` as an SPM dependency, and xcodebuild throws a similar error. Can you try if other SPM dependencies work for you? If not, then I doubt that it's an issue on our end, but instead an issue with Xcode.

**Comment by 0ax1** [comment]
I managed to pin this one down in the meantime. The minimal example projects, including the one using `sentry-cocoa` which I provided, finish their archive builds successfully on the command-line if either `-scheme` or `-targeet` is passed to `xcodebuild`.

Further, `Skip Install` has to be set to `No` and a valid install path needs to be picked for the build target in Xcode, in order for the archive builds to succeed.
